feat: Add COPY IDMS support #704

schte01 opened the pull request:
Signed-off-by: Teri Schnitt <teri.schnitt@broadcom.com>

Added COPY IDMS support to Data Division, Working-Storage/Linkage sections and Procedure Division.

temanbrcom commented on server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolParser.g4:
```diff
@@ -791,10 +830,13 @@ procedureDivisionByValue
    : DECLARATIVES DOT_FS procedureDeclarative+ END DECLARATIVES DOT_FS
    ;
-
+//procedureDeclarative
+//   : procedureSectionHeader DOT_FS useStatement DOT_FS paragraphs
+//   ;
```
Please, remove the commented-out code

temanbrcom commented on server/src/test/java/com/broadcom/lsp/cobol/usecases/TestReplacingWithDifferentPatternsShowsError.java:
```diff
@@ -48,9 +48,10 @@
+  private static final String MESSAGE = "Missing token IDMS at dataDescriptionEntryCopyIdms";
+
+  private static final String MESSAGE2 =
+      "Syntax error on 'REPLACING' expected {<EOF>, ASSOCIATED-DATA, ASSOCIATED-DATA-LENGTH, BOUNDS, CAPABLE, CCSVERSION, CICS, CLOSE-DISPOSITION, COMMITMENT, CONVENTION, CRUNCH, DEFAULT-DISPLAY, DFHRESP, DFHVALUE, DISK, ENCODING, EXITS, FUNCTION-POINTER, ID, IDENTIFICATION, IGNORED, IMPLICIT, IN, INVOKED, KEPT, KEYBOARD, LIBRARY, LINKAGE, LOCAL-STORAGE, LONG-DATE, LONG-TIME, LOWER, LTERM, LR, MA ...
```
Please, split the line not to extend 120 characters length

temanbrcom commented on server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolLexer.g4:
```diff
@@ -285,6 +289,7 @@ SHIFT_IN : S H I F T MINUSCHAR I N;
 SHIFT_OUT : S H I F T MINUSCHAR O U T;
 SHORT_DATE : S H O R T MINUSCHAR D A T E;
 SIGN : S I G N;
+SIZE : S I Z E;
```
SIZE is already defined in Db2SqlLexer.g4

temanbrcom commented on server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolLexer.g4:
```diff
@@ -164,6 +164,7 @@ HIGH_VALUE : H I G H MINUSCHAR V A L U E;
 HIGH_VALUES : H I G H MINUSCHAR V A L U E S;
 ID : I D;
 IDENTIFICATION : I D E N T I F I C A T I O N;
+IDMS : I D M S;
```
Please, add the new tokens to the `cobolWord` rule

schte01 commented:
I have made the requested changes and amended the commit.

On Wed, Feb 3, 2021 at 9:42 AM Andrei Temnikov <notifications@github.com>
wrote:

> *@temanbrcom* requested changes on this pull request.
> ------------------------------
>
> In
> server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolParser.g4
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r569459841>
> :
>
> > +//procedureDeclarative
> +//   : procedureSectionHeader DOT_FS useStatement DOT_FS paragraphs
> +//   ;
>
> Please, remove the commented-out code
> ------------------------------
>
> In
> server/src/test/java/com/broadcom/lsp/cobol/usecases/TestReplacingWithDifferentPatternsShowsError.java
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r569461238>
> :
>
> > @@ -48,9 +48,10 @@
>        "         NAME2.\n" + "           DISPLAY \"ABC\".\n" + "           MOVE 0 TO PARENT. ";
>    private static final String NEW2_NAME = "NEW2";
>
> -  private static final String MESSAGE =
> -      "Syntax error on 'COPY' expected {<EOF>, ID, IDENTIFICATION, LINKAGE, LOCAL-STORAGE, WORKING-STORAGE, PROCEDURE, "
> -          + "SCHEMA, END, FILE, MAP, SQL, 'EXEC SQL', EXEC, '01-49', '66', '77', '88'}";
> +  private static final String MESSAGE = "Missing token IDMS at dataDescriptionEntryCopyIdms";
> +
> +  private static final String MESSAGE2 =
> +      "Syntax error on 'REPLACING' expected {<EOF>, ASSOCIATED-DATA, ASSOCIATED-DATA-LENGTH, BOUNDS, CAPABLE, CCSVERSION, CICS, CLOSE-DISPOSITION, COMMITMENT, CONVENTION, CRUNCH, DEFAULT-DISPLAY, DFHRESP, DFHVALUE, DISK, ENCODING, EXITS, FUNCTION-POINTER, ID, IDENTIFICATION, IGNORED, IMPLICIT, IN, INVOKED, KEPT, KEYBOARD, LIBRARY, LINKAGE, LOCAL-STORAGE, LONG-DATE, LONG-TIME, LOWER, LTERM, LR, MAPS, NATIONAL-EDITED, NETWORK, NODENAME, NOWRITE, NUMERIC-DATE, NUMERIC-TIME, ODT, ORDERLY, OWN, POINTER-32, PORT, PRINTER, PTERM, READER, RECEIVED, RECURSIVE, REDEFINES, REMOTE, SAVE, SCREENSIZE, SHORT-DATE, SIZE, SUBSCHEMA-AREANAMES, SUBSCHEMA-CONTROL, SUBSCHEMA-CTRL, SUBSCHEMA-DESCRIPTION, SUBSCHEMA-DML-LR-DESCRIPTION, SUBSCHEMA-LR-CONTROL, SUBSCHEMA-LR-CTRL, SUBSCHEMA-LR-DESCRIPTION, SUBSCHEMA-LR-NAMES, SUBSCHEMA-LR-RECORDS, SUBSCHEMA-NAMES, SUBSCHEMA-RECNAMES, SUBSCHEMA-RECORDS, SUBSCHEMA-SETNAMES, SUBSCHEMA-NAME, SYSVERSION, THREAD, THREAD-LOCAL, TODAYS-DATE, TODAYS-NAME, TRUNCATED, TYPEDEF, UTF-8, VIRTUAL, WORKING-STORAGE, ABSOLUTE, ACCELERATION, ACCELERATOR, ACTIVATE, ACTIVE, ADA, AGE, ALIAS, ALLOW, ALTERIN, ALWAYS, APPEND, APPLCOMPAT, APPLICATION, ARCHIVE, ASC, ASCII, ASSERTION, ASSOCIATE, ASUTIME, ATOMIC, AUTHID, AUTHORIZATION, AUTOMATIC, AUX, AVG, BEGIN, BETWEEN, BIGINT, BIND, BINDADD, BIT-LENGTH, BLOB, BLOCKED, BOTH, BUFFERPOOL, BUFFERPOOLS, BUSINESS-TIME, CACHE, CALLED, CALLER, CAPTURE, CARDINALITY, CASCADE, CASE, CAST, CATALOG, CATALOG-NAME, CHANGED, CHANGES, CHAR-LENGTH, CHARACTER-LENGTH, CHECKED, CLAUSE, CLOB, CLUSTER, COBOL, CODEUNITS32, COLLATE, COLLATION, COLLECT, COLLECTION, COLLID, COLUMN, COMMENT, COMMIT, COMMITTED, COMPARISONS, CONCAT, CONCENTRATE, CONCURRENT, CONDITION-NUMBER, CONNECTION, CONNECTION-NAME, CONSERVATIVE, CONSTRAINT, CONSTRAINTS, CONTEXT, CONVERT, CORRELATION, COUNT, COUNT-BIG, CPU, CREATEIN, CREATETAB, CROSS, CS, CUBE, CURRENT-DATE, CURRENT-PATH, CURRENT-SCHEMA, CURRENT-SERVER, CURRENT-SQLID, CURRENT-TIME, CURRENT-TIMESTAMP, CURRENT-TIMEZONE, CURRENT-USER, CURRENTLY, CURSOR, CURSOR-NAME, CURSORS, CYCLE, DATABASE, DATALINK, DB2-AUTHENTICATION-TYPE, DB2-AUTHORIZATION-ID, DB2-CONNECTION-STATE, DB2-CONNECTION-STATUS, DB2-ENCRYPTION-TYPE, DB2-ERROR-CODE1, DB2-ERROR-CODE2, DB2-ERROR-CODE3, DB2-ERROR-CODE4, DB2-GET-DIAGNOSTICS-DIAGNOSTICS, DB2-INTERNAL-ERROR-POINTER, DB2-LAST-ROW, DB2-LINE-NUMBER, DB2-MESSAGE-ID, DB2-MODULE-DETECTING-ERROR, DB2-NUMBER-PARAMETER-MARKERS, DB2-NUMBER-RESULT-SETS, DB2-NUMBER-ROWS, DB2-ORDINAL-TOKEN1, DB2-ORDINAL-TOKEN2, DB2-ORDINAL-TOKEN3, DB2-ORDINAL-TOKEN4, DB2-PRODUCT-ID, DB2-REASON-CODE, DB2-RETURN-STATUS, DB2-RETURNED-SQLCODE, DB2-ROW-NUMBER, DB2-SERVER-CLASS-NAME, DB2-SQL-ATTR-CURSOR-HOLD, DB2-SQL-ATTR-CURSOR-ROWSET, DB2-SQL-ATTR-CURSOR-SCROLLABLE, DB2-SQL-ATTR-CURSOR-SENSITIVITY, DB2-SQL-ATTR-CURSOR-TYPE, DB2-SQL-NESTING-LEVEL, DB2-SQLERRD1, DB2-SQLERRD2, DB2-SQLERRD3, DB2-SQLERRD4, DB2-SQLERRD5, DB2-SQLERRD6, DB2-SQLERRD-SET, DB2-TOKEN-COUNT, DB2DARI, DB2GENERAL, DB2GENRL, DB2SQL, DBADM, DBCLOB, DBINFO, DEADLOCKS, DEALLOCATE, DEBUG, DEC, DEC-ROUND-CEILING, DEC-ROUND-DOWN, DEC-ROUND-FLOOR, DEC-ROUND-HALF-DOWN, DEC-ROUND-HALF-EVEN, DEC-ROUND-HALF-UP, DEC-ROUND-UP, DECFLOAT, DECIMAL, DECLARE, DEFAULT, DEFAULTS, DEFER, DEFERRABLE, DEFERRED, DEFINEBIND, DEFINERUN, DEFINITION, DEGREE, DESC, DESCRIBE, DESCRIPTOR, DETERMINISTIC, DIAGNOSTICS, DIMENSIONS, DISALLOW, DISPATCH, DISTINCT, DO, DOMAIN, DOUBLE, DOUBLEQUOTE, DROP, DROPIN, DYNAMICRULES, EACH, EBCDIC, ELIGIBLE, ELSEIF, EMPTY, ENVIRONMENT, ESCAPE, EUR, EXACT, EXCEPT, EXCHANGE, EXCLUDE, EXCLUDING, EXCLUSIVE, EXISTS, EXIT, EXPLAIN, EXTENDED, EXTENSION, FAILBACK, FEDERATED, FENCED, FETCH, FINAL, FIRST, FLOAT, FLUSH, FOLLOWING, FOREIGN, FORMAT, FORTRAN, FOUND, FREEPAGE, FS, FULL, GBPCACHE, GENERAL, GENERATED, GET-ACCEL-ARCHIVE, GOTO, GRANT, GRAPHIC, GROUP, GROUPING, HANDLER, HAVING, HOUR, IDENTITY, IMPLICIT-SCHEMA, INCLUDING, INDEXES, INDICATOR, INHERIT, INITIAL-INSTS, INITIAL-IOS, INITIALLY, INNER, INOUT, INSENSITIVE, INSTEAD, INSTS-PER-ARGBYTE, INSTS-PER-INVOC, INT, INTEGER, INTEGRITY, INTERSECT, INVALID, INVOKEBIND, INVOKERUN, IOS-PER-ARGBYTE, IOS-PER-INVOC, ISO, ISOLATION, ITERATE, JAVA, JIS, JOBNAME, JOIN, KEYS, LABELS, LANGUAGE, LARGE, LEAVE, LIKE, LIMIT, LINKTYPE, LITERALS, LOCAL, LOCATION, LOCATOR, LOCATORS, LOCKED, LOCKS, LOCKSIZE, LOGGED, LONG, LONGVAR, LOOP, MAINTAINED, MAPPING, MASK, MATCH, MAX, MAXVALUE, MESSAGE-TEXT, MICROSECOND, MICROSECONDS, MINUTE, MINVALUE, MIXED, MODIFIES, MODULE, MONTHS, NAMED, NAMES, NATIONAL, NATURAL, NCHAR, NEW, NEW-TABLE, NEXTVAL, NICKNAME, NOCACHE, NOCYCLE, NODE, NOMINVALUE, NOORDER, NULLABLE, NUMBER, OBJECT, OCTETS, OF, OFFSET, OLD, OLD-TABLE, OLE, OLEDB, ONCE, ONLINE, ONLY, OPTHINT, OPTIMIZATION, OPTIMIZE, OUT, OUTCOME, OUTER, OVER, OVERLAPS, OVERRIDING, PACKAGE, PAD, PARALLEL, PARAMETER, PART, PARTIAL, PARTITION, PARTITIONING, PASCAL, PASSTHRU, PCTFREE, PERCENT-ARGBYTES, PERMISSION, PIECESIZE, PIPE, PLAN, PLI, PORTION, PRECEDING, PRECISION, PRESERVE, PRIMARY, PRIOR, PRIQTY, PRIVILEGES, PROCEDURE, PROTOCOL, PUBLIC, QUALIFIER, QUERYNO, RANGE, READS, REAL, RECOMMEND, RECOVERY, REF, REFERENCING, REFRESH, REGISTERS, RENAME, REOPT, REPEAT, REPLICATED, REQUIRED, RESIDENT, RESIGNAL, RESOLUTION, RESOLVE, RESTORE, RESTRICT, RESULT-SET-LOCATOR, RETURN-STATUS, RETURNED-SQLSTATE, RETURNS, REVOKE, ROLLUP, ROUNDING, ROUTINE, ROW, ROW-COUNT, ROWID, ROWS, ROWSET, RR, RS, SAVEPOINT, SBCS, SCALE, SCHEMA, SCRATCHPAD, SCROLL, SECOND, SECQTY, SECURED, SELECTIVITY, SELF, SENSITIVE, SEQUENCE, SERIALIZABLE, SERVER-NAME, SESSION-USER, SETS, SHARE, SHRLEVEL, SIGNAL, SIMPLE, SMALLINT, SNAPSHOT, SOME, SPECIAL, SPECIFIC, SQLCA, SQLCODE, SQLDA, SQLERROR, SQLEXCEPTION, SQLID, SQLSTATE, SQLWARNING, STABILIZED, STACKED, STARTING, STATEMENT, STATEMENTS, STATIC, STATISTICS, STATS, STAY, STMTCACHE, STMTID, STMTTOKEN, STOGROUP, STORED, STORES, STRUCTURE, STYLE, SUB, SUBSTR, SUBSTRING, SUMMARY, SWITCH, SYNONYM, SYSTEM, SYSTEM-TIME, SYSTEM-USER, TABLE, TABLE-NAME, TABLESPACE, TABLESPACES, TEMPORARY, THREADSAFE, TIMESTAMP, TIMEZONE, TIMEZONE-HOUR, TIMEZONE-MINUTE, TRANSLATE, TRANSLATION, TREAT, TRIM, TRUSTED, UNBOUNDED, UNDER, UNDO, UNICODE, UNION, UNIQUE, UNKNOWN, UNPACK, UPPER, USA, USER, VALIDATE, VARBINARY, VARCHAR, VARGRAPHIC, VARIABLE, VARIANT, VCAT, VERSION, VIEW, VOLATILE, WAITFORDATA, WHENEVER, WHERE, WHILE, WITHOUT, WLM, WORK, WRAPPER, YEARS, YES, ZONE, ABCODE, ABDUMP, ABEND, ABORT, ABPROGRAM, ABSTIME, ACCUM, ACEE, ACQACTIVITY, ACQPROCESS, ACQUACTIVITY, ACTION, ACTIVITY, ACTIVITYID, ACTPARTN, AID, ALARM, ALLOCATE, ALTSCRNHT, ALTSCRNWD, ANYKEY, APLKYBD, APLTEXT, APPLID, AS, ASA, ASKTIME, ASRAINTRPT, ASRAKEY, ASRAPSW, ASRAREGS, ASRASPC, ASRASTG, ASYNCHRONOUS, ATTACH, ATTACHID, ATTRIBUTES, AUTHENTICATE, AUTOPAGE, AUXILIARY, BASE64, BASICAUTH, BELOW, BIF, BINARY, BIT, BODYCHARSET, BOOKMARK, BRDATA, BRDATALENGTH, BREXIT, BRIDGE, BROWSETOKEN, BTRANS, BUFFER, BUILD, BURGEABILITY, CADDRLENGTH, CARD, CBUFF, CCSID, CERTIFICATE, CHANGE, CHANGETIME, CHANNEL, CHAR, CHARACTERSET, CHECK, CHUNKEND, CHUNKING, CHUNKNO, CHUNKYES, CICSDATAKEY, CIPHERS, CLEAR, CLICONVERT, CLIENT, CLIENTADDR, CLIENTADDRNU, CLIENTCONV, CLIENTNAME, CLNTADDR6NU, CLNTIPFAMILY, CLOSESTATUS, CLRPARTN, CMDSEC, CNAMELENGTH, CNOTCOMPL, CODE, CODEPAGE, COLOR, COMMAREA, COMMONNAME, COMMONNAMLEN, COMPAREMAX, COMPAREMIN, COMPLETE, COMPOSITE, COMPSTATUS, CONDITION, CONFIRM, CONFIRMATION, CONNECT, CONSISTENT, CONSOLE, CONTAINER, CONTEXTTYPE, CONVDATA, CONVERSE, CONVERTST, CONVERTTIME, CONVID, COPY, COUNTER, COUNTRY, COUNTRYLEN, CREATE, CRITICAL, CTLCHAR, CURRENT, CWA, CWALENG, DATA1, DATA2, DATALENGTH, DATALENTH, DATAONLY, DATAPOINTER, DATASTR, DATATOXML, DATATYPE, DATCONTAINER, DATEFORM, DATESEP, DATESTRING, DAYCOUNT, DAYOFMONTH, DAYOFWEEK, DAYOFYEAR, DAYS, DAYSLEFT, DCOUNTER, DDMMYY, DDMMYYYY, DEBKEY, DEBREC, DEEDIT, DEFINE, DEFRESP, DEFSCRNHT, DEFSCRNWD, DELAY, DELETEQ, DEQ, DESTCOUNT, DESTID, DESTIDLENG, DETAILLENGTH, DIGEST, DIGESTTYPE, DISCONNECT, DOCDELETE, DOCSIZE, DOCSTATUS, DOCTOKEN, DOCUMENT, DS3270, DSSCS, DUMP, DUMPCODE, DUMPID, ECADDR, ECBLIST, EIB, ELEMNAME, ELEMNAMELEN, ELEMNS, ELEMNSLEN, END, ENDACTIVITY, ENDBR, ENDBROWSE, ENDFILE, ENDOUTPUT, ENQ, ENTER, ENTRYNAME, EODS, EPRFIELD, EPRFROM, EPRINTO, EPRLENGTH, EPRSET, EPRTYPE, ERASE, ERASEAUP, ERRTERM, ESMREASON, ESMRESP, EVENT, EVENTTYPE, EVENTUAL, EWASUPP, EXPECT, EXPIRYTIME, EXTDS, EXTRACT, FACILITY, FACILITYTOKN, FAULTACTLEN, FAULTACTOR, FAULTCODE, FAULTCODELEN, FAULTCODESTR, FAULTSTRING, FAULTSTRLEN, FCI, FCT, FIELD, FILE, FIRESTATUS, FLENGTH, FMH, FMHPARM, FORCE, FORMATTIME, FORMFEED, FORMFIELD, FREE, FREEKB, FREEMAIN, FROMACTIVITY, FROMCCSID, FROMCHANNEL, FROMCODEPAGE, FROMDOC, FROMLENGTH, FROMPROCESS, FRSET, FULLDATE, GCHARS, GCODES, GDS, GENERIC, GET, GETMAIN, GETNEXT, GMMI, GROUPID, GTEC, GTEQ, HANDLE, HEAD, HEADER, HEX, HILIGHT, HOLD, HONEOM, HOST, HOSTCODEPAGE, HOSTLENGTH, HOSTTYPE, HOURS, HTTPHEADER, HTTPMETHOD, HTTPRNUM, HTTPVERSION, HTTPVNUM, IGNORE, IMMEDIATE, INCREMENT, INCLUDE, INITIMG, INITPARM, INITPARMLEN, INPARTN, INPUTEVENT, INPUTMSG, INPUTMSGLEN, INQUIRE, INSERT, INTERVAL, INTO, INTOCCSID, INTOCODEPAGE, INVALIDCOUNT, INVITE, INVOKINGPROG, ISSUE, ISSUER, ITEM, IUTYPE, JOURNALNAME, JTYPEID, JUSFIRST, JUSLAST, JUSTIFY, KATAKANA, KEEP, KEYLENGTH, KEYNUMBER, L40, L64, L80, LANGINUSE, LANGUAGECODE, LASTUSETIME, LDC, LDCMNEM, LDCNUM, LENGTHLIST, LEVEL, LIGHTPEN, LINK, LIST, LISTLENGTH, LLID, LOAD, LOCALITY, LOCALITYLEN, LOGMESSAGE, LOGMODE, LOGONLOGMODE, LOGONMSG, LUNAME, MAIN, MAP, MAPCOLUMN, MAPHEIGHT, MAPLINE, MAPONLY, MAPPED, MAPPINGDEV, MAPSET, MAPWIDTH, MASSINSERT, MAXDATALEN, MAXFLENGTH, MAXIMUM, MAXLENGTH, MAXLIFETIME, MAXPROCLEN, MCC, MEDIATYPE, MESSAGEID, METADATA, METADATALEN, METHOD, METHODLENGTH, MILLISECONDS, MINIMUM, MINUTES, MMDDYY, MMDDYYYY, MODENAME, MONITOR, MONTH, MONTHOFYEAR, MSR, MSRCONTROL, NAME, NAMELENGTH, NATLANG, NATLANGINUSE, NETNAME, NEWPASSWORD, NEWPHRASE, NEWPHRASELEN, NEXTTRANSID, NLEOM, NOAUTOPAGE, NOCC, NOCHECK, NOCLICONVERT, NOCLOSE, NODATA, NODOCDELETE, NODUMP, NOEDIT, NOFLUSH, NOHANDLE, NOINCONVERT, NONE, NOOUTCONERT, NOQUEUE, NOQUIESCE, NOSRVCONVERT, NOSUSPEND, NOTE, NOTPURGEABLE, NOTRUNCATE, NOWAIT, NSCONTAINER, NUMCIPHERS, NUMEVENTS, NUMITEMS, NUMREC, NUMROUTES, NUMSEGMENTS, NUMTAB, OIDCARD, OPCLASS, OPERATION, OPERATOR, OPERID, OPERKEYS, OPERPURGE, OPID, OPSECURITY, OPTIONS, ORGABCODE, ORGANIZATLEN, ORGUNIT, ORGUNITLEN, OUTDESCR, OUTLINE, OUTPARTN, OWNER, PA, PAGENUM, PAGING, PARTN, PARTNER, PARTNPAGE, PARTNS, PARTNSET, PASS, PASSBK, PASSWORD, PASSWORDLEN, PATH, PATHLENGTH, PCT, PFXLENG, PHRASE, PHRASELEN, PIPLENGTH, PIPLIST, POINT, POOL, POP, PORTNUMBER, PORTNUMNU, POST, PPT, PREDICATE, PREFIX, PREPARE, PRINCONVID, PRINSYSID, PRINT, PRIORITY, PRIVACY, PROCESS, PROCESSTYPE, PROCLENGTH, PROCNAME, PROFILE, PROGRAM, PROTECT, PS, PUNCH, PURGEABLE, PUSH, PUT, QNAME, QUERY, QUERYPARM, QUERYSTRING, QUERYSTRLEN, RBA, RBN, READNEXT, READPREV, READQ, REATTACH, RECEIVER, RECFM, RECORD, RECORDLEN, RECORDLENGTH, REDUCE, REFPARMS, REFPARMSLEN, RELATESINDEX, RELATESTYPE, RELATESURI, REMOVE, REPEATABLE, REPETABLE, REPLY, REPLYLENGTH, REQID, REQUESTTYPE, RESCLASS, RESETBR, RESID, RESIDLENGTH, RESOURCE, RESP, RESP2, RESSEC, RESTART, RESTYPE, RESULT, RESUME, RETAIN, RETCODE, RETCORD, RETRIECE, RETRIEVE, RETURNPROG, RIDFLD, ROLE, ROLELENGTH, ROLLBACK, ROUTE, ROUTECODES, RPROCESS, RRESOURCE, RRN, RTERMID, RTRANSID, SADDRLENGTH, SCHEME, SCHEMENAME, SCOPE, SCOPELEN, SCRNHT, SCRNWD, SECONDS, SEGMENTLIST, SENDER, SERIALNUM, SERIALNUMLEN, SERVER, SERVERADDR, SERVERADDRNU, SERVERCONV, SERVERNAME, SESSION, SESSTOKEN, SHARED, SIGDATA, SIGNOFF, SIGNON, SIT, SNAMELENGTH, SOAPFAULT, SOSI, SPOOLCLOSE, SPOOLOPEN, SPOOLREAD, SPOOLWRITE, SRVCONVERT, SRVRADDR6NU, SRVRIPFAMILY, SSLTYPE, STARTBR, STARTBROWSE, STARTCODE, STATE, STATELEN, STATIONID, STATUSCODE, STATUSLEN, STATUSTEXT, STORAGE, STRINGFORMAT, SUBADDR, SUBCODELEN, SUBCODESTR, SUBEVENT, SUSPEND, SUSPSTATUS, SYMBOL, SYMBOLLIST, SYNCHRONOUS, SYNCLEVEL, SYNCONRETURN, SYNCPOINT, SYSID, TABLES, TASK, TASKPRIORITY, TCPIP, TCPIPSERVICE, TCT, TCTUA, TCTUALENG, TD, TELLERID, TEMPLATE, TERMCODE, TERMID, TERMPRIORITY, TEXTKYBD, TEXTLENGTH, TEXTPRINT, TIMEOUT, TIMER, TIMESEP, TOACTIVITY, TOCHANNEL, TOCONTAINER, TOFLENGTH, TOKEN, TOLENGTH, TOPROCESS, TRACE, TRACENUM, TRAILER, TRANPRIORITY, TRANSACTION, TRANSFORM, TRANSID, TRIGGER, TRT, TS, TWA, TWALENG, TYPENAME, TYPENAMELEN, TYPENS, TYPENSLEN, UNATTEND, UNCOMMITTED, UNESCAPED, UNLOCK, SQL, 'EXEC SQL', EXEC, EXECUTE, UOW, UPDATE, URI, URIMAP, URL, URLLENGTH, USERDATAKEY, USERID, USERNAME, USERNAMELEN, USERPRIORITY, VALIDATION, VALUELENGTH, VERIFY, VERSIONLEN, VOLUME, VOLUMELENG, WAIT, WAITCICS, WEB, WPMEDIA, WRAP, WRITEQ, WSACONTEXT, WSAEPR, XCTL, XMLCONTAINER, XMLTODATA, XMLTRANSFORM, XRBA, YEAR, YYDDD, YYDDMM, YYYYDDD, YYYYDDMM, YYYYMMDD, ZERO-DIGITAL, '.', '=', '01-49', '66', '77', '88', IDENTIFIER}";
>
> Please, split the line not to extend 120 characters length
> ------------------------------
>
> In server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolLexer.g4
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r569468041>
> :
>
> > @@ -285,6 +289,7 @@ SHIFT_IN : S H I F T MINUSCHAR I N;
>  SHIFT_OUT : S H I F T MINUSCHAR O U T;
>  SHORT_DATE : S H O R T MINUSCHAR D A T E;
>  SIGN : S I G N;
> +SIZE : S I Z E;
>
> SIZE is already defined in Db2SqlLexer.g4
> ------------------------------
>
> In server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CobolLexer.g4
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r569468520>
> :
>
> > @@ -164,6 +164,7 @@ HIGH_VALUE : H I G H MINUSCHAR V A L U E;
>  HIGH_VALUES : H I G H MINUSCHAR V A L U E S;
>  ID : I D;
>  IDENTIFICATION : I D E N T I F I C A T I O N;
> +IDMS : I D M S;
>
> Please, add the new tokens to the cobolWord rule
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#pullrequestreview-582424386>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQITZIAL4WLW47LELSHIJR3S5FOD3ANCNFSM4XA6DFOA>
> .
>

-- 
This electronic communication and the information and any files transmitted 
with it, or attached to it, are confidential and are intended solely for 
the use of the individual or entity to whom it is addressed and may contain 
information that is confidential, legally privileged, protected by privacy 
laws, or otherwise restricted from disclosure to anyone else. If you are 
not the intended recipient or the person responsible for delivering the 
e-mail to the intended recipient, you are hereby notified that any use, 
copying, distributing, dissemination, forwarding, printing, or copying of 
this e-mail is strictly prohibited. If you received this e-mail in error, 
please return the e-mail to the sender, delete it from your computer, and 
destroy any printed copy of it.

grianbrcom reviewed (approved):
:zebra:

temanbrcom reviewed (approved):
🖌️

temanbrcom commented on server/src/test/java/com/broadcom/lsp/cobol/usecases/TestReplacingWithDifferentPatternsShowsError.java:
```diff
@@ -48,9 +48,153 @@
+  private static final String MESSAGE = "Missing token IDMS at dataDescriptionEntryCopyIdms";
+
+  private static final String MESSAGE2 =
+          "Syntax error on 'REPLACING' expected {<EOF>, ASSOCIATED-DATA, ASSOCIATED-DATA-LENGTH, BOUNDS, CAPABLE," +
```
Please, apply auto-formatting for this class to pass the style check

grianbrcom replied:
In IDEA IDE you can enable `Operation sign on next line` option:
![image](https://user-images.githubusercontent.com/60617592/106878669-7febde00-66da-11eb-9474-24351df824fe.png)

schte01 commented:
I have made the formatting change to the test file.

On Thu, Feb 4, 2021 at 4:58 AM Andrei Temnikov <notifications@github.com>
wrote:

> *@temanbrcom* commented on this pull request.
> ------------------------------
>
> In
> server/src/test/java/com/broadcom/lsp/cobol/usecases/TestReplacingWithDifferentPatternsShowsError.java
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r570091448>
> :
>
> > @@ -48,9 +48,153 @@
>        "         NAME2.\n" + "           DISPLAY \"ABC\".\n" + "           MOVE 0 TO PARENT. ";
>    private static final String NEW2_NAME = "NEW2";
>
> -  private static final String MESSAGE =
> -      "Syntax error on 'COPY' expected {<EOF>, ID, IDENTIFICATION, LINKAGE, LOCAL-STORAGE, WORKING-STORAGE, PROCEDURE, "
> -          + "SCHEMA, END, FILE, MAP, SQL, 'EXEC SQL', EXEC, '01-49', '66', '77', '88'}";
> +  private static final String MESSAGE = "Missing token IDMS at dataDescriptionEntryCopyIdms";
> +
> +  private static final String MESSAGE2 =
> +          "Syntax error on 'REPLACING' expected {<EOF>, ASSOCIATED-DATA, ASSOCIATED-DATA-LENGTH, BOUNDS, CAPABLE," +
>
> Please, apply auto-formatting for this class to pass the style check
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#pullrequestreview-583217577>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQITZICAPSUERJSTXDEXDRDS5JVSTANCNFSM4XA6DFOA>
> .
>

-- 
This electronic communication and the information and any files transmitted 
with it, or attached to it, are confidential and are intended solely for 
the use of the individual or entity to whom it is addressed and may contain 
information that is confidential, legally privileged, protected by privacy 
laws, or otherwise restricted from disclosure to anyone else. If you are 
not the intended recipient or the person responsible for delivering the 
e-mail to the intended recipient, you are hereby notified that any use, 
copying, distributing, dissemination, forwarding, printing, or copying of 
this e-mail is strictly prohibited. If you received this e-mail in error, 
please return the e-mail to the sender, delete it from your computer, and 
destroy any printed copy of it.

schte01 commented:
Can you elaborate on what this setting does? Thank you.

On Thu, Feb 4, 2021 at 5:18 AM Anton Grigorev <notifications@github.com>
wrote:

> *@grianbrcom* commented on this pull request.
> ------------------------------
>
> In
> server/src/test/java/com/broadcom/lsp/cobol/usecases/TestReplacingWithDifferentPatternsShowsError.java
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r570105359>
> :
>
> > @@ -48,9 +48,153 @@
>        "         NAME2.\n" + "           DISPLAY \"ABC\".\n" + "           MOVE 0 TO PARENT. ";
>    private static final String NEW2_NAME = "NEW2";
>
> -  private static final String MESSAGE =
> -      "Syntax error on 'COPY' expected {<EOF>, ID, IDENTIFICATION, LINKAGE, LOCAL-STORAGE, WORKING-STORAGE, PROCEDURE, "
> -          + "SCHEMA, END, FILE, MAP, SQL, 'EXEC SQL', EXEC, '01-49', '66', '77', '88'}";
> +  private static final String MESSAGE = "Missing token IDMS at dataDescriptionEntryCopyIdms";
> +
> +  private static final String MESSAGE2 =
> +          "Syntax error on 'REPLACING' expected {<EOF>, ASSOCIATED-DATA, ASSOCIATED-DATA-LENGTH, BOUNDS, CAPABLE," +
>
> In IDEA IDE you can enable Operation sign on next line option:
> [image: image]
> <https://user-images.githubusercontent.com/60617592/106878669-7febde00-66da-11eb-9474-24351df824fe.png>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#discussion_r570105359>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQITZIF64K4TTPVQGMQFAI3S5JX57ANCNFSM4XA6DFOA>
> .
>

-- 
This electronic communication and the information and any files transmitted 
with it, or attached to it, are confidential and are intended solely for 
the use of the individual or entity to whom it is addressed and may contain 
information that is confidential, legally privileged, protected by privacy 
laws, or otherwise restricted from disclosure to anyone else. If you are 
not the intended recipient or the person responsible for delivering the 
e-mail to the intended recipient, you are hereby notified that any use, 
copying, distributing, dissemination, forwarding, printing, or copying of 
this e-mail is strictly prohibited. If you received this e-mail in error, 
please return the e-mail to the sender, delete it from your computer, and 
destroy any printed copy of it.

grianbrcom commented:
Sure, if you have a string like that:
```
  String str = "Some string. >< Other text.";
```
And want to split it in two, you can put the cursor between `><` and press enter. Then IDEA converts it into:
```
  String str = "Some string. >" 
    + "< Other text.";
```
If you have `Operation sign on next line` option selected. Otherwise it will be like this:
```
  String str = "Some string. >" +
    "< Other text.";
```

schte01 commented:
Thank you.

On Thu, Feb 4, 2021 at 5:42 AM Anton Grigorev <notifications@github.com>
wrote:

> Sure, if you have a string like that:
>
>   String str = "Some string. >< Other text.";
>
> And want to split it in two, you can put the cursor between >< and press
> enter. Then IDEA converts it into:
>
>   String str = "Some string. >"
>     + "< Other text.";
>
> If you have Operation sign on next line option selected. Otherwise it
> will be like this:
>
>   String str = "Some string. >" +
>     "< Other text.";
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/che-che4z-lsp-for-cobol/pull/759#issuecomment-773210789>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQITZIFHTVK24CWZBXGGP3TS5J2XVANCNFSM4XA6DFOA>
> .
>

-- 
This electronic communication and the information and any files transmitted 
with it, or attached to it, are confidential and are intended solely for 
the use of the individual or entity to whom it is addressed and may contain 
information that is confidential, legally privileged, protected by privacy 
laws, or otherwise restricted from disclosure to anyone else. If you are 
not the intended recipient or the person responsible for delivering the 
e-mail to the intended recipient, you are hereby notified that any use, 
copying, distributing, dissemination, forwarding, printing, or copying of 
this e-mail is strictly prohibited. If you received this e-mail in error, 
please return the e-mail to the sender, delete it from your computer, and 
destroy any printed copy of it.
